src/toolkits/drawing_classifier/drawing_classifier.cpp

```diff
@@ -700,14 +700,28 @@ gl_sarray drawing_classifier::predict(gl_sframe data, std::string output_type) {
     result = result.apply(max_prob_label, class_labels.front().get_type());
 
   } else if (output_type == "probability") {
-    auto max_prob = [=](const flexible_type& ft) {
+    /** The output_type="probability" is to provide the probability of the True
+     *  class in binary classifiers.
+     *  For example, if one is building a binary classifier for
+     *  "cat" vs "not cat", output_type="probability" should output the
+     *  probability of the data point being "cat", not the probability of the
+     *  predicted class.
+     */
+
+    size_t num_classes = read_state<flex_int>("num_classes");
+    DASSERT_GT(num_classes, 0);
+
+    if (num_classes > 2) {
+      log_and_throw(
+          "Use probability_vector in case of multi-class classification.");
+    }
+
+    auto true_class_probability = [=](const flexible_type& ft) {
       const flex_vec& prob_vec = ft.get<flex_vec>();
-      auto max_it = std::max_element(prob_vec.begin(), prob_vec.end());
-      return *max_it;
+      return prob_vec.back();
```

Comment:
I think we need to align with the label order generated from the training dataset.

Reply:
Could you sync with @hoytak? We should align with what all the supervised-learning toolkits do.

Reply:
This should align with the label order in the training set that are sorted according to class name. So for example:

```
model.classes -> ['cat', 'dog', 'foosa']
```
and predictions should be in that order
```
[p_cat, p_dog, p_foosa
```

Sadly the binary classifier is a little annoying because all other packages have the convention of using a "reference" class to minimize multi-collinearity and have fewer parameters while this isn't much of a concept in deep learning based models.

Reply:
@hoytak , @Jarvi-Izana , and I discussed this at length and here are our thoughts:

1. For `output_type="probability"` in a binary classifier setting, all our supervised learning toolkits currently follow the approach of sorting labels lexicographically, and returning the probability of the last label.
2. This approach is kind of arbitrary, since the labels could be "cat" and "not cat", in which case the user will end up with probabilities for the label, "not cat".
3. After some discussion, we came to the conclusion that for TC 6.0, we should modify this behavior across Drawing Classifier, Logistic Classifier, SVMs, and other supervised learning toolkits to the following: `output_type="probability"` should only apply if the target values in the data are `0` and `1`. This issue is tracked in #2667 .

Here is what I suggest w.r.t this PR:

1. Let's merge this PR in its current state where it matches what we currently have for all supervised learning toolkits.
2. Let's address #2667 before TC 6.0 and make a code change in this logic (and elsewhere) then.

@nickjong @srikris @Jarvi-Izana @hoytak Thoughts?

Reply:
As long as it's book-kept, I'm good.

Reply:
I don't think we have the time to adequately design, develop, and test this kind of change for 6.0.